stdlib/public/core/Integers.swift.gyb

```diff
@@ -77,56 +77,912 @@ infix operator &>>= : AssignmentPrecedence
+  ///   or both have bits set to `1`.
+""",
+        '^': """\
+  /// Returns the result of performing a bitwise XOR operation on this value
+  /// and the given value.
+  ///
+  /// A bitwise XOR operation, also known as an exclusive OR operation, results
+  /// in a value that has each bit set to `1` where *one or the other but not
+  /// both* of its arguments had that bit set to `1`. For example:
+  ///
+  ///     let x: UInt8 = 5          // 0b00000101
+  ///     let y: UInt8 = 14         // 0b00001110
+  ///     let z = x.bitwiseXor(y)   // 0b00001011
+  ///
+  /// This method serves as the basis for the bitwise XOR operator (`^`).
+  ///
+  /// - Parameter rhs: The value to perform the bitwise XOR with this value.
+  /// - Returns: A new value with each bit set to `1` where either this value
+  ///   or `rhs`, but not both, have bits set to `1`.
+""",
+
+        '&>>': """\
+  /// Returns the result of shifting this value's binary representation the
+  /// specified number of digits to the right.
+  ///
+  /// In a *masking shift*, the value passed as `rhs` is masked to produce a
+  /// value in the range `0..<x.bitWidth`. The shift is performed using this
+  /// masked value.
+  ///
+  /// For example, if you pass `2` to the `maskingShiftRight(_:)` method on an
+  /// 8-bit, unsigned integer, the shift amount requires no masking.
+  ///
+  ///     let x: UInt8 = 30                 // 0b00011110
+  ///     let y = x.maskingShiftRight(2)
+  ///     // y == 7                         // 0b00000111
+  ///
+  /// However, if you pass `11` as `rhs`, the method first bitmasks `rhs` to
+  /// `3`, and then uses that masked value as the number of bits to shift `x`.
+  ///
+  ///     let z = x.maskingShiftRight(11)
+  ///     // z == 3                         // 0b00000011
+  ///
+  /// This method serves as the basis for the masking right shift operator
+  /// (`&>>`).
+  ///
+  ///     let y3 = x &>> 11
+  ///     // y3 == 3
+  ///
+  /// Relationship to the Right Shift Operator
+  /// ----------------------------------------
+  ///
+  /// The `maskingShiftRight(_:)` method handles attempted overshifts and
+  /// undershifts differently from the right shift operator (`>>`). When the
+  /// value passed `rhs` in a masking shift is within the range
+  /// `0...<bitWidth`, the operation is equivalent to using the right shift
+  /// operator.
+  ///
+  ///     let x: UInt8 = 30                 // 0b00011110
+  ///     let y1 = x.maskingShiftRight(2)
+  ///     // y1 == 7                        // 0b00000111
+  ///     let y2 = x >> 2
+  ///     // y2 == 7                        // 0b00000111
+  ///
+  /// The right shift operator does not mask its right-hand-side argument, so
+  /// passing `11` as `rhs` shifts all the bits of `x` to zero.
+  ///
+  ///     let z1 = x.maskingShiftRight(11)
+  ///     // z1 == 240                      // 0b00000011
+  ///     let z2 = x >> 11
+  ///     // z2 == 0                        // 0b00000000
+  ///
```

Comment:
I understand what you're trying to say.  I'm saying it doesn't communicate well.  The text should say that in a masking shift, the shift amount's bit representation is masked before it is applied so that the actual shift amount is always between zero and the number of bits in the lhs.  Communicating it with one line of example code is not sufficient, IMO.  It should also say that masking shifts require more care to use correctly, but are likely to be more efficient when used with shift amounts that are not compile-time constants, compiling down to a single instruction on most architectures.